### ISSUE TITLE
Increase importance of layout rules or include them after components

### ISSUE THREAD

**Issue by pwolfert** [CMSgov/design-system#781]
**Description**

We're seeing a slightly different behavior when applying layout utility classes to elements with component classes as a result of the layout package being consolidated into the new core package. For example, in one of our products (DM me for an example screenshot), we have a list element with three list items with layout classes that make it so they are rendered as three columns on larger screens and stacked on smaller screens, like so:

```html
<ul class="ds-c-list ds-c-list--bare ds-l-row">
  <li class="ds-l-col--12 ds-l-sm-col--4">...</li>
  <li class="ds-l-col--12 ds-l-sm-col--4">...</li>
  <li class="ds-l-col--12 ds-l-sm-col--4">...</li>
</ul>
```

We use the `ds-c-list` and `ds-c-list--bare` to strip away the default list styles. Unfortunately, these rules now take precedence over the layout styles, so the CSS rules get applied in the following way, where the last rule has the highest priority:

```css
.ds-l-row, .ds-l-form-row {
    display: -ms-flexbox;
    display: flex;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
}
.ds-l-row {
    margin-left: -16px;
    margin-right: -16px;
}
.ds-c-list {
    margin-bottom: 16px;
    margin-top: 16px;
    padding-left: 32px;
}
.ds-c-list--bare {
    list-style: none;
    margin: 0;
    padding: 0;
}
```

Where we wanted the row to have `-16px` padding on either side, it ends up having `0px` padding.

I think that since layout classes are utility classes that are applied directly by the developer, they should always take precedence over component styles. I know that could possibly break other people's apps if before v2 they were importing the layout package first and the core second, so this should be a discussion.

**Steps to reproduce the issue**

Use the above HTML with some content in the list items to see the issue.

**Comment by line47** [comment]
Thanks for pointing this out @pwolfert - I'm going to look into this and report back here.

Could you drop your screenshot in the issue here?

**Comment by pwolfert** [comment]
I can't post a screenshot of the app because of confidentiality, but I can share it with you directly.

**Comment by bernardwang** [comment]
We plan to address this issue by regrouping grid classes under utilities. Originally these classes were under the `design-system-settings` package, so when we combined it we put grid alongside our HTML element classes like `ds-base`, `ds-h1`, `ds-text` etc, but you are right, it is basically used as a utility and should take precedence. Moving it to utilities would also be consistent with how [USWDS organizes](https://designsystem.digital.gov/utilities/layout-grid/) it.


Tracked here: https://jira.cms.gov/browse/WNMGDS-658

**Comment by bernardwang** [comment]
Fixed in [v2.2.1](https://github.com/CMSgov/design-system/releases/tag/core-2.2.1)
